src/py/pyodide/http.py

```diff
@@ -301,9 +313,19 @@ async def pyfetch(url: str, **kwargs: Any) -> FetchResponse:
     {'info': {'arch': 'wasm32', 'platform': 'emscripten_3_1_32',
     'version': '0.23.4', 'python': '3.11.2'}, ... # long output truncated
     """
+    controller = AbortController.new()
+    if "signal" in kwargs:
+        kwargs["signal"] = AbortSignal.any([kwargs["signal"], controller.signal])
```

Comment:
[`AbortSignal.any` seems quite new](https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal/any_static), Chrome started to support it in v116 (Released in 2023-08-15), Firefox v124 (2024-03-19), and Node.js v20.3.

So it would be nice if we can achieve this without using any. Personally, I would like to support Node.js v18 until its end-of-life (April 2025).

Reply:
Thanks for suggestion. Let me try polyfill it in python then

Reply:
I found a few posts about userland solutions, which may be helpful to you.

- https://github.com/shaseley/abort-signal-any/blob/main/README.md#userland-solutions
- https://samthor.au/2022/abortcontroller-is-your-friend/ (Helpers that may or may not exist section)

Reply:
Thank you very much🌹

Reply:
You can do copy paste a js polyfill i'd recommend.

Reply:
Yeah I will do this tomorrow

Reply:
Sadly maybe not that simple. [The userland solution](https://github.com/shaseley/abort-signal-any/blob/main/README.md#userland-solutions) will leak memory since it don't remove the event listeners.

However, since we don't need to implement a perfect polyfill for `AbortSignal`, we just need to use it in our `pyfetch`, whose lifespan is under our control.

I plan to remove these event listeners in a `__del__`, when all JsFetchResponse are deleted.

Reply:
I had a look of [this document](https://docs.google.com/document/d/1LvmsBLV85p-PhSGvTH-YwgD6onuhh1VXLg8jPlH32H4) and noticed that it is actually polyfillable if only the browser supports weak references. But I failed to find a polyfill that implements this (possibly because I lack experience in searching for polyfills, in fact I simply did a quick search on npm).

Reply:
We assume weak references and finalization registry so a polyfill should be possible.

Reply:
[Here](https://github.com/pyodide/pyodide/pull/4846/commits/52387f009a0f0edf23eda76a3f055adb401408fd#diff-1fb6404e8153dfd2db40ec8a5d07af6de9a4a5d596710907980397c79204d3aa) it is ~